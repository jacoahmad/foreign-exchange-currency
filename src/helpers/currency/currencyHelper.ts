import CurrencyISO from "../../static/iso4217.json";
import { AVAILABLE_CURRENCIES } from "./currencyConstant";

export function calculateRatesByBaseValue(
  rates: { [key: string]: any },
  value: number
) {
  const calculatedRates: { [key: string]: any } = {};
  const baseRate: { [key: string]: any } = { ...rates };
  Object.keys(baseRate).forEach((rate: string) => {
    calculatedRates[rate] = Number(baseRate[rate]) * value;
  });
  return calculatedRates;
}

export function toISOCurrencyName(currencyCode: string) {
  return ((CurrencyISO as { [key: string]: any })[currencyCode] || {}).name;
}

export type CurrencyOption = {
  label: string;
  value: string;
};

export type CurrencyOptions = CurrencyOption[];

export const currencyOptions: CurrencyOptions = AVAILABLE_CURRENCIES.map(
  currency => ({
    value: currency,
    label: currency
  })
);
