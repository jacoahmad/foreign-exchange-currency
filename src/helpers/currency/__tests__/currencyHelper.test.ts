import * as helpers from '../currencyHelper';
import mockRates from './mockData.json';

describe('calculateRates', () => {

  it('Calculate rates based at it values', () => {
    expect(helpers.calculateRatesByBaseValue(mockRates, 10)).toMatchSnapshot();
  });

  it('Returns correct Currency Name', () => {
    // THB = Thai baht;
    const currencyName = helpers.toISOCurrencyName('THB');
    expect(currencyName).toBe('Thai baht');
  });
  
})