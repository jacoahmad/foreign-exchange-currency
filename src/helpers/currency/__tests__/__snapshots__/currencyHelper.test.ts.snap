// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateRates Calculate rates based at it values 1`] = `
Object {
  "AUD": 14.48038176,
  "BGN": 17.2834924,
  "BRL": 38.314775539,
  "CAD": 13.191940615,
  "CHF": 9.815305762,
  "CNY": 68.74513962500001,
  "CZK": 226.307882644,
  "DKK": 65.974726052,
  "EUR": 8.837044892,
  "GBP": 7.902527395,
  "HKD": 78.140685755,
  "HRK": 65.405620361,
  "HUF": 2862.937433722,
  "IDR": 141277.83669141,
  "ILS": 36.041887593,
  "INR": 695.462177448,
  "ISK": 1250.441852245,
  "JPY": 1074.938140686,
  "KRW": 11615.500176741,
  "MXN": 190.613290916,
  "MYR": 41.535878402,
  "NOK": 85.595616826,
  "NZD": 15.253623188,
  "PHP": 514.784376105,
  "PLN": 37.631671969,
  "RON": 41.745316366000004,
  "RUB": 632.017497349,
  "SEK": 93.94485684,
  "SGD": 13.575468363,
  "THB": 308.050547897,
  "TRY": 58.153057618000005,
  "USD": 10,
  "ZAR": 143.317426653,
}
`;
